### PR TITLE
Reorganize structure

### PR DESCRIPTION
**Pull request by quiqua**
Organize by feature so that each feature can be injected with its own
components and doesn't need to bother with other components if they are not
relevant.

### CODE REVIEW

File: .idea/misc.xml [hunk 1]
```diff
@@ -25,7 +25,7 @@
       </value>
     </option>
   </component>
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_7" project-jdk-name="1.8" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" project-jdk-name="1.8" project-jdk-type="JavaSDK">
```
Comment (elaydis) [inline .idea/misc.xml]: don't commit files in .idea 😛

File: .idea/saveactions_settings.xml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (elaydis) [inline .idea/saveactions_settings.xml]: same here 😛

File: app/src/main/kotlin/eu/quiqua/gloomhaven/statstracker/app/App.kt [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package eu.quiqua.gloomhaven.statstracker.app
+
+import android.app.Application
+import eu.quiqua.gloomhaven.statstracker.app.dagger.AppComponent
+import eu.quiqua.gloomhaven.statstracker.app.dagger.AppModule
+import eu.quiqua.gloomhaven.statstracker.app.dagger.DaggerAppComponent
+
+class App : Application() {
```
Comment (elaydis) [inline app/src/main/kotlin/eu/quiqua/gloomhaven/statstracker/app/App.kt]: why `App` not `Application`?

File: app/src/main/kotlin/eu/quiqua/gloomhaven/statstracker/app/dagger/AppComponent.kt [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+import android.content.Context
+import android.content.SharedPreferences
+import dagger.Component
+import eu.quiqua.gloomhaven.statstracker.app.App
+import javax.inject.Singleton
+
+@Singleton
+@Component(modules = [(AppModule::class)])
+interface AppComponent {
```
Comment (elaydis) [inline app/src/main/kotlin/eu/quiqua/gloomhaven/statstracker/app/dagger/AppComponent.kt]: `ApplicationComponent` ?! (I don't like shortening names 😛 )

File: app/src/main/kotlin/eu/quiqua/gloomhaven/statstracker/features/stats/dagger/StatsModule.kt [hunk 5]
```diff
@@ -0,0 +1,12 @@
+package eu.quiqua.gloomhaven.statstracker.features.stats.dagger
+
+import dagger.Module
+import dagger.Provides
+
+@Module
+class StatsModule() {
+
+    @Provides
+    fun getModel(): String = "hello world"
```
Comment (elaydis) [inline app/src/main/kotlin/eu/quiqua/gloomhaven/statstracker/features/stats/dagger/StatsModule.kt]: in our team we call these methods `providesXyz` - doesn't matter much though 🤷‍♀️

File: app/src/main/kotlin/eu/quiqua/gloomhaven/statstracker/app/App.kt (hunk 3 above)
Comment (elaydis) [reply app/src/main/kotlin/eu/quiqua/gloomhaven/statstracker/app/App.kt]: I would go for something like `StatsTrackerApplication` actually :)

**Comment by quiqua** [comment]
Thanks @elaydis :)

**Pull request merged**